File: client-app/src/app/models/activity.ts
import { Profile } from "./profile"

export interface Activity {
  id: string
  title: string
  date: Date | null
  description: string
  category: string
  city: string
  venue: string
  hostUsername: string
  isCancelled: boolean
  attendees: Profile[]
  host?: Profile
  isHost: boolean
  isGoing: boolean
}
export class Activity implements Activity{
  constructor(init?: ActivityFormValues){
    Object.assign(this, init);
  }
}

export class ActivityFormValues {
  id?: string = undefined;
  title: string = '';
  description: string = '';
  category: string = '';
  city: string = '';
  venue: string = '';
  date: Date | null = null;

  constructor(activity?: ActivityFormValues) {
    if (activity) {
      this.id = activity.id;
      this.title = activity.title;
      this.description = activity.description;
      this.category = activity.category;
      this.city = activity.city;
      this.venue = activity.venue;
      this.date = activity.date;
      
    }

  }
}

File: Domain/Activity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Activity
    {
        public Guid ID {get;set;}
        public string Title{get;set;}
        public DateTime Date{get;set;}
        public string Description {get;set;}
        public string Category{get;set;}
        public string City{get;set;}
        public string Venue{get;set;}

        public Boolean isCancelled {get;set;}

        public ICollection<Comment> Comments {get;set;} = new List<Comment>();

        public ICollection<ActivityAttendee> Attendees{get;set;} = new List<ActivityAttendee>();

    }
}